src/components/singlePlayerHomepage/SinglePlayerHomepage.jsx

```diff
@@ -1,5 +1,79 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+
 import "./css/singlePlayerHomepageStyle.css";
+import QuizCard from "../quizCard/QuizCard";
+import RulesModal from "../rulesModal/RulesModal";
+import { ErrorToast } from "../toasts/Toasts";
+import Loader from "../loader/Loader";
+import { useQuiz } from "../../contexts/quizContext";
 
 export default function SinglePlayerHomepage() {
-  return <h2>Single Player Homepage</h2>;
+  const [hasUserSelected, setHasUserSelected] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(true);
+  const [caughtError, setCaughtError] = useState(null);
+  const [allQuizes, setAllQuizes] = useState([]);
+  const { quiz, setQuiz } = useQuiz();
+
+  async function getAllQuizes() {
+    setCaughtError(null);
+    setIsLoaded(false);
+    try {
+      const res = await axios.get(
+        "https://b-mock-qz.vercel.app/api/get-quizes"
+      );
+      setAllQuizes(res.data.quizes);
+      setIsLoaded(true);
```

Comment:
set state only when you get response status as success status, cause this code will set state even on error status

Reply:
I took care of that on the server-side.